### PR TITLE
add rounded logo option to header

### PR DESCRIPTION
**Pull request by DimaGolt**
Added rounded factory constructor for home header.
ezgif-2-e964e3a3f1

### CODE REVIEW

File: lib/features/greeting_screen/greeting_screen.dart [hunk 1]
```diff
@@ -15,6 +17,7 @@ class GreetingScreen extends StatefulWidget {
 
 class _GreetingScreenState extends State<GreetingScreen> {
   final TextEditingController _codeController = TextEditingController();
+  bool isHomeHeader = true;
```
Comment (mikolaj-jalocha) [inline lib/features/greeting_screen/greeting_screen.dart:20]: This could be final I guess

**Review by mikolaj-jalocha — COMMENTED** [review]
Like @simon-the-shark good job with initial sliver header code :-) In my opinion clean way here of dealing with 2 types of headers. 

I just left one note regarding final variable :-)

File: lib/features/greeting_screen/greeting_screen.dart (hunk 1 above)
Comment (simon-the-shark) [reply lib/features/greeting_screen/greeting_screen.dart:20]: not really :P

Comment (simon-the-shark) [reply lib/features/greeting_screen/greeting_screen.dart:20]: he uses `setState` below and changes it

Comment (mikolaj-jalocha) [reply lib/features/greeting_screen/greeting_screen.dart:20]: Sorry, didn't notice

**Review by simon-the-shark — COMMENTED** [review]
Good code. I'm sure you know this, but using `method`s and `get`ters like this is not 100% optimal, cause the calculations in there are performed few times each build instead of one.
 
But I get it's not a performance issue, we do there like a one or two simple arithmetics, so won't see a real world difference (unless you have half a thread cpu and 2kb of ram)

So if you thinks it adds to readability and it was a conscious choice of yours, then go for it. Just showing you this if haven't thought of simply just this:
```dart
@override
  Widget build(BuildContext context, double shrinkOffset, bool overlapsContent) {
    final bgBottomMargin = clampDouble(expandedHeight * 0.3 - shrinkOffset, 0, double.infinity);

    return Stack(
// ...
```
instead of this
```dart
 double _bgBottomMargin(double shrinkOffset) =>
      clampDouble(expandedHeight * 0.3 - shrinkOffset, 0, double.infinity);

  @override
  Widget build(BuildContext context, double shrinkOffset, bool overlapsContent) {
    return Stack(
// ...
```

and you can basically do this for each of your methods or getters. When you move the calcs back to the build method and assign it to variable, then maybe you will get those few splits of milliseconds :P - cause you perform each calculation only once a build and not once a usage.

But as I said, you won't see a difference with your eye. If you think yours more readable, then you can leave it this way. Just wanted to make sure you've considered this option too. In my opinion the readability is similar. You can take inspiration or not. Take care and happy coding / reading through this lengthy comment

File: lib/widgets/logo_header.dart [hunk 2]
```diff
@@ -24,85 +24,72 @@ class LogoHeader extends SliverPersistentHeaderDelegate {
...
-  double _bgBottomMargin(double shrinkOffset) =>
-      clampDouble(expandedHeight * 0.3 - shrinkOffset, 0, double.infinity);
+    double logoSize = _isRounded
+        ? MediaQuery.of(context).size.width * 0.4
+        : MediaQuery.of(context).size.width / 2;
+
+    //Background related
+    double bgHeight = expandedHeight * 0.7;
+
+    double bgBottomMargin = clampDouble(expandedHeight * 0.3 - shrinkOffset, 0, double.infinity);
```
Comment (simon-the-shark) [inline lib/widgets/logo_header.dart:55]: All of these variables can be `final` :)

**Review by simon-the-shark — CHANGES_REQUESTED** [review]

**Review by simon-the-shark — APPROVED** [review]

**Pull request merged**